File: lesson-03/js/ex08.js
let countRec = 0;
function fibRec(n) {
    countRec += 1;
    return (n === 1 || n === 2)
        ? 1
        : fibRec(n - 1) + fibRec(n - 2);
}

let countFor = 0;
function fibFor(n) {
    if (n === 1 || n === 2) return 1;
    let a1 = 1, a2 = 1;
    let aN = 0;
    for (let i = 2; i < n; i++) {
        countFor += 1;
        aN = a2 + a1;
        a1 = a2;
        a2 = aN;
    }
    return aN;
}

for (let n = 1; n < 30; n++) {
    countRec = 0;
    countFor = 0;
    console.log(`fibRec(${n}) ${fibRec(n)} countRec ${countRec}`);
    console.log(`fibFor(${n}) ${fibFor(n)} countFor ${countFor} \n`);

}

File: lesson-03/java/ex08.java
public class ex08 {
    static int countRec = 0;
    static int countFor = 0;

    public static void main(String[] args) {
        for (int n = 1; n < 30; n++) {
            countRec = 0;
            countFor = 0;
            System.out.printf("fibRec(%d) %d countRec = %d\n", n, fibRec(n), countRec);
            System.out.println("fibFor(" + n + ") " + fibFor(n) + " countFor " + countFor + "\n");
        }
    }

    public static int fibRec(int n) {
        countRec++;
        return (n == 1 || n == 2) ? 1 : fibRec(n - 1) + fibRec(n - 2);
    }

    public static int fibFor(int n) {
        if (n == 1 || n == 2)
            return 1;
        int a1 = 1, a2 = 1;
        int aN = 0;
        for (int i = 2; i < n; i++) {
            countFor++;
            aN = a2 + a1;
            a1 = a2;
            a2 = aN;
        }
        return aN;
    }
}
